lib/diag.rs

```diff
@@ -85,7 +93,7 @@ impl DiagMap {
     // OSD or not.
     // `osd_diag` holds an OSD's removability status. Using a binary heap we
     // can always know which state it has that holds the highest precedent.
-    pub fn exhaustive_diag(self) -> Status {
+    pub fn exhaustive_diag(self, json_format: bool) -> Status {
```

Comment:
I'd also be tempted to refactor this to have the various diags return an object that we can then output in main() to reduce repetitive rendering :) maybe the struct would ```impl render(Format)```?